packages/providers/test/provider.test.ts

```diff
@@ -708,17 +720,90 @@ describe('Provider', () => {
+    resetCache();
+
+    expect(Provider.chainInfoCache[FUEL_NETWORK_URL]).toBeUndefined();
+    expect(Provider.nodeInfoCache[FUEL_NETWORK_URL]).toBeUndefined();
+
+    await Provider.create(FUEL_NETWORK_URL);
+
+    expect(Provider.chainInfoCache[FUEL_NETWORK_URL]).toBeDefined();
+    expect(Provider.nodeInfoCache[FUEL_NETWORK_URL]).toBeDefined();
+  });
+
+  it('should ensure getChain and getNode uses the cache and does not fetch new data', async () => {
+    resetCache();
+
+    const spyFetchChainAndNodeInfo = jest.spyOn(Provider.prototype, 'fetchChainAndNodeInfo');
+    const spyFetchChain = jest.spyOn(Provider.prototype, 'fetchChain');
+    const spyFetchNode = jest.spyOn(Provider.prototype, 'fetchNode');
+
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+
+    expect(spyFetchChainAndNodeInfo).toHaveBeenCalledTimes(1);
+    expect(spyFetchChain).toHaveBeenCalledTimes(1);
+    expect(spyFetchNode).toHaveBeenCalledTimes(1);
+
+    provider.getChain();
+    provider.getNode();
+
+    expect(spyFetchChainAndNodeInfo).toHaveBeenCalledTimes(1);
+    expect(spyFetchChain).toHaveBeenCalledTimes(1);
+    expect(spyFetchNode).toHaveBeenCalledTimes(1);
+  });
+
+  it('should ensure fetchChainAndNodeInfo always fetch new data', async () => {
+    resetCache();
 
-    const provider1 = await Provider.create(FUEL_NETWORK_URL);
-    const provider2 = await Provider.create(FUEL_NETWORK_URL);
+    const spyFetchChainAndNodeInfo = jest.spyOn(Provider.prototype, 'fetchChainAndNodeInfo');
+    const spyFetchChain = jest.spyOn(Provider.prototype, 'fetchChain');
+    const spyFetchNode = jest.spyOn(Provider.prototype, 'fetchNode');
 
-    // `getChainInfoWithoutInstance` should only be called once, we reuse the cached value for the second provider
-    expect(spyGetChainInfo).toHaveBeenCalledTimes(1);
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+
+    expect(spyFetchChainAndNodeInfo).toHaveBeenCalledTimes(1);
+    expect(spyFetchChain).toHaveBeenCalledTimes(1);
+    expect(spyFetchNode).toHaveBeenCalledTimes(1);
+
+    await provider.fetchChainAndNodeInfo();
+
+    expect(spyFetchChainAndNodeInfo).toHaveBeenCalledTimes(2);
+    expect(spyFetchChain).toHaveBeenCalledTimes(2);
+    expect(spyFetchNode).toHaveBeenCalledTimes(2);
+  });
+
+  it('should ensure getGasConfig return essential gas related data', async () => {
+    const provider = await Provider.create(FUEL_NETWORK_URL);
+
+    const gasConfig = provider.getGasConfig();
+
+    expect(gasConfig.gasPerByte).toBeDefined();
+    expect(gasConfig.gasPriceFactor).toBeDefined();
+    expect(gasConfig.maxGasPerPredicate).toBeDefined();
+    expect(gasConfig.maxGasPerTx).toBeDefined();
+    expect(gasConfig.minGasPrice).toBeDefined();
+  });
 
-    expect(provider1.url).toEqual(FUEL_NETWORK_URL);
-    expect(provider2.url).toEqual(FUEL_NETWORK_URL);
+  it('should throws when using getChain or getNode and without cached data', async () => {
```

Comment:
Given that `getChain` and `getNode` are instance methods that are only accessible via a `Provider` instance that is only ever instantiated via `Provider.create`, it seems to me that the setup of this test is a forced example which should never happen. @Dhaiwat10 is it really necessary for `getChain` and `getNode` to throw these errors?